Lib9c/Model/Quest/QuestList.cs

```diff
@@ -0,0 +1,386 @@
+                    case GeneralQuestSheet.Row row2:
+                        quest = new GeneralQuest(row2, reward);
+                        _quests.Add(quest);
+                        break;
+                    case ItemEnhancementQuestSheet.Row row3:
+                        quest = new ItemEnhancementQuest(row3, reward);
+                        _quests.Add(quest);
+                        break;
+                    case ItemGradeQuestSheet.Row row4:
+                        quest = new ItemGradeQuest(row4, reward);
+                        _quests.Add(quest);
+                        break;
+                    case MonsterQuestSheet.Row row5:
+                        quest = new MonsterQuest(row5, reward);
+                        _quests.Add(quest);
+                        break;
+                    case TradeQuestSheet.Row row6:
+                        quest = new TradeQuest(row6, reward);
+                        _quests.Add(quest);
+                        break;
+                    case WorldQuestSheet.Row row7:
+                        quest = new WorldQuest(row7, reward);
+                        _quests.Add(quest);
+                        break;
+                    case ItemTypeCollectQuestSheet.Row row8:
+                        quest = new ItemTypeCollectQuest(row8, reward);
+                        _quests.Add(quest);
+                        break;
+                    case GoldQuestSheet.Row row9:
+                        quest = new GoldQuest(row9, reward);
+                        _quests.Add(quest);
+                        break;
+                    case CombinationEquipmentQuestSheet.Row row10:
+                        int stageId;
+                        var recipeRow = equipmentItemRecipeSheet.Values
+                            .FirstOrDefault(r => r.Id == row10.RecipeId);
+                        if (recipeRow is null)
+                        {
+                            throw new ArgumentException($"Invalid Recipe Id : {row10.RecipeId}");
+                        }
+
+                        stageId = recipeRow.UnlockStage;
+                        quest = new CombinationEquipmentQuest(row10, reward, stageId);
+                        _quests.Add(quest);
+                        break;
+                }
+            }
+        }
+
+
+        public QuestList(Dictionary serialized)
+        {
+            _listVersion = serialized.TryGetValue((Text) "list-version", out var listVersion)
+                ? listVersion.ToInteger()
+                : 1;
+            
+            _quests = serialized.TryGetValue((Text) "quests", out var questsValue)
+                ? questsValue.ToList(Quest.Deserialize)
+                : new List<Quest>();
+
+            completedQuestIds = serialized.TryGetValue((Text) "completedQuestIds", out var idsValue)
+                ? idsValue.ToList(StateExtensions.ToInteger)
+                : new List<int>();
+        }
+
+        public void UpdateList(
```

Comment:
i think it would be better add version parameter when updateList called.  because check the version information at the time of the update.